consensus/state_processing/src/per_epoch_processing/single_pass.rs

```diff
@@ -342,20 +345,66 @@ pub fn process_epoch_single_pass<E: EthSpec>(
     // Finish processing pending balance deposits if relevant.
     //
     // This *could* be reordered after `process_pending_consolidations` which pushes only to the end
-    // of the `pending_balance_deposits` list. But we may as well preserve the write ordering used
+    // of the `pending_deposits` list. But we may as well preserve the write ordering used
     // by the spec and do this first.
-    if let Some(ctxt) = pending_balance_deposits_ctxt {
-        let mut new_pending_balance_deposits = List::try_from_iter(
+    if let Some(ctxt) = pending_deposits_ctxt {
+        let mut new_balance_deposits = List::try_from_iter(
             state
-                .pending_balance_deposits()?
+                .pending_deposits()?
                 .iter_from(ctxt.next_deposit_index)?
                 .cloned(),
         )?;
         for deposit in ctxt.deposits_to_postpone {
-            new_pending_balance_deposits.push(deposit)?;
+            new_balance_deposits.push(deposit)?;
         }
-        *state.pending_balance_deposits_mut()? = new_pending_balance_deposits;
+        *state.pending_deposits_mut()? = new_balance_deposits;
         *state.deposit_balance_to_consume_mut()? = ctxt.deposit_balance_to_consume;
+
+        // Apply the new deposits to the state
+        for deposit in ctxt.new_validator_deposits.into_iter() {
+            if is_valid_deposit_signature(
+                &DepositData {
+                    pubkey: deposit.pubkey,
+                    withdrawal_credentials: deposit.withdrawal_credentials,
+                    amount: deposit.amount,
+                    signature: deposit.signature,
+                },
+                spec,
+            )
+            .is_ok()
+            {
+                let mut validator = Validator {
```

Comment:
this will also be significantly simplified now that we have `add_validator_to_registry()` :D